### DISCUSSION TITLE
`flux create source helm` doesn't work

### DISCUSSION THREAD

**Discussion by agardnerIT** [fluxcd/flux2#3235] [Q&A] [answered]
Followed the [getting started](https://fluxcd.io/flux/get-started/) guide on kind and it works without Helm.

Now trying to follow the [Flux for Helm Users](https://fluxcd.io/flux/use-cases/helm/) page:

`flux create source helm traefik --url https://helm.traefik.io/traefik --namespace traefik` doesn't seem to work.

```
flux create source helm traefik --url https://helm.traefik.io/traefik --namespace traefik
✚ generating HelmRepository source
► applying HelmRepository source
✔ source created
◎ waiting for HelmRepository source reconciliation
✔ HelmRepository source reconciliation completed
✔ fetched revision: 1af279fb087c31c617d203239ad810526a86b38138dc89c496fe189f0f5c90e4
```

When I `flux get sources all`:

```
NAME                     	        REVISION          	SUSPENDED	 READY                                                                        
gitrepository/flux-system	main/98ddb86  	False    	         True
gitrepository/podinfo    	master/1cf228c	False    	         True 
```

```
flux version        
flux: v0.35.0
helm-controller: v0.25.0
kustomize-controller: v0.29.0
notification-controller: v0.27.0
source-controller: v0.30.0

flux check                                                                               
► checking prerequisites
✔ Kubernetes 1.25.0 >=1.20.6-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.25.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.29.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.27.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.30.0
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta1
✔ buckets.source.toolkit.fluxcd.io/v1beta2
✔ gitrepositories.source.toolkit.fluxcd.io/v1beta2
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta1
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1beta2
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta1
✔ receivers.notification.toolkit.fluxcd.io/v1beta1
✔ all checks passed
```

**Comment by agardnerIT** [comment, marked as answer]
Edit: Re-opening this discussion as: `flux create source helm traefik --url https://helm.traefik.io/traefik --namespace traefik`

Doesn't seem to work and definitely doesn't create any new files on disk. `git status` is clean.


~Ignore me. My fault.~

**Reply by stefanprodan** [reply]
You need to pass `--export > traefik-repo.yaml` and the command will create the object on disk instead of creating in on cluster. Then push the file to the Git repo and Flux will apply in on the cluster for you.

To list the Helm repositories, you need to specify the namespace:

```sh
flux get sources all -n traefik
```

Or list them at cluster level:

```sh
flux get sources all --all-namespaces
```

Please see here how to organise your Flux repo: https://github.com/fluxcd/flux2-kustomize-helm-example

**Comment by agardnerIT** [comment]
Thanks, I got it figured out. The confusion came because the docs page reads:

```
The main difference is that the Flux client will not imperatively create resources in the cluster. Instead, these commands create Custom Resource files, which are committed to version control as instructions only (note: you may use the --export flag to manage any file edits with finer grained control before pushing to version control)
```

Whereas of course, if you **do not** use --export it **seems** to create things imperatively without files - hence the confusion.

So perhaps this discussion could frame a good first issue to clarify the wording of the docs. Side note that I find it strange for a GitOps tool to do things "behind the scenes" without commitable files. Logically, one would probably expect everything to be represented as files. Just my 2c.
